src/inventory/transaction/action/DropItemAction.php

```diff
@@ -45,6 +45,9 @@ public function validate(Player $source) : void{
 		if($this->targetItem->getCount() > $this->targetItem->getMaxStackSize()){
 			throw new TransactionValidationException("Target item exceeds item type max stack size");
 		}
+		if($this->targetItem->getLockMode() !== null){
+			throw new TransactionValidationException("Target item is locked in inventory or slot");
+		}
```

Comment:
I don't think this check is necessary. If the origin item is actually locked in a slot, the SlotChangeAction changing it to air will be invalid anyway.

Reply:
This is necessary because without this validation, target item can be dropped when target item is locked in inventory.